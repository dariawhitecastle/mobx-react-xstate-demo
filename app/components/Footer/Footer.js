import React from 'react'
import { FormattedMessage } from 'react-intl'

import { LocaleToggle } from 'containers/LocaleToggle'
import { Anchor } from '../Anchor'
import Wrapper from './Wrapper'
import messages from './messages'

function Footer() {
  return (
    <Wrapper>
      <section>
        <FormattedMessage {...messages.licenseMessage} />
      </section>
      <section>
        <LocaleToggle />
      </section>
      <section>
        <FormattedMessage
          {...messages.authorMessage}
          values={{
            author: <Anchor href="https://twitter.com/mxstbr">Max Stoiber</Anchor>,
          }}
        />
      </section>
    </Wrapper>
  )
}

export default Footer
