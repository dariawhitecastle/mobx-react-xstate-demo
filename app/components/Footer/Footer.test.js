import React from 'react'
import renderer from 'react-test-renderer'
import { IntlProvider } from 'react-intl'
import { Provider } from 'mobx-react'
import { RouterStore } from 'mobx-react-router'

import Footer from './Footer'
import trunk from '../../configureStore'

describe('<Footer />', () => {
  let stores

  beforeAll(() => {
    const routingStore = new RouterStore()
    stores = {
      routing: routingStore,
    }
    trunk.init()
  })

  it('should render and match the snapshot', () => {
    const renderedComponent = renderer
      .create(
        <Provider {...stores}>
          <IntlProvider locale="en">
            <Footer />
          </IntlProvider>
        </Provider>,
      )
      .toJSON()

    expect(renderedComponent).toMatchSnapshot()
  })
})
