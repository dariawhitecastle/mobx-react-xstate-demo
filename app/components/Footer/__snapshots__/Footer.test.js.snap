// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render and match the snapshot 1`] = `
<footer
  className="Wrapper-wcfdfo-0 dPvBog"
>
  <section>
    <span>
      This project is licensed under the MIT license.
    </span>
  </section>
  <section>
    <div
      className="styles__Wrapper-gw8d9h-0 hMaTPt"
    >
      <select
        className="Select-sc-1sm01tk-0 wIjFa"
        onChange={[Function]}
        value="en"
      >
        <option
          value="en"
        >
          en
        </option>
        <option
          value="de"
        >
          de
        </option>
      </select>
    </div>
  </section>
  <section>
    <span>
      
      Made with love by 
      <a
        className="StyledAnchor-sc-1rp7lwl-0 jgeOSx"
        href="https://twitter.com/mxstbr"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Max Stoiber
      </a>
      .
    
    </span>
  </section>
</footer>
`;
